tests/test_views.py

```diff
@@ -1,15 +1,32 @@
 import pytest
 
 
-def test_api_parse_succeds(client):
-    # TODO: Finish this test. Send a request to the API and confirm that the
-    # data comes back in the appropriate format.
+def test_api_parse_succeeds(client):
+    '''Test that the API parses a valid address string successfully and returns data in the correct structure.'''
+    
     address_string = '123 main st chicago il'
-    pytest.fail()
+    
+    response = client.get('/api/parse/', {'address': address_string })
+    
+    assert response.data == {
+        'inputString': address_string,
+        'components': {
+            'AddressNumber':        '123',
+            'StreetName':           'main',
+            'StreetNamePostType':   'st',
+            'PlaceName':            'chicago',
+            'StateName':            'il',
+        }, 
+        'type': 'Street Address'
+    }
 
 
 def test_api_parse_raises_error(client):
-    # TODO: Finish this test. The address_string below will raise a
-    # RepeatedLabelError, so ParseAddress.parse() will not be able to parse it.
+    '''Test that the API returns an exception when parsing an invalid string string successfully and returns the correct exception name.'''
+    
     address_string = '123 main st chicago il 123 main st'
-    pytest.fail()
+    
+    response = client.get('/api/parse/', {'address': address_string})
+    # Could this call ^ be fixturized for DRYer code? 
```

Comment:
Technically, yes! You could use [parameterization](https://github.com/datamade/testing-guidelines/blob/master/intermediate-python-testing.md#parameterizing-fixtures) to pull the address string out into a variable. Here's [an example in the wild](https://github.com/datamade/la-metro-councilmatic/blob/23e1dccf8c60f7cf2c3f5624c05b28b233b7e25c/tests/test_events.py#L59-L93) of conditionally performing test assertions, depending on parameter values. 

I think whether you'd want to combine success and failure cases is a matter of preference and context. It isn't always the best separation of concerns, although [pytest's docs on params include an example with mixed outcomes](https://docs.pytest.org/en/stable/example/parametrize.html#parametrizing-conditional-raising), but it can be a good strategy for reducing duplicate code if, as in these tests, most of the logic is the same between test cases.

I'll leave it to you whether you want to take a stab at it!

Reply:
Thanks! Making a wild guess that the majority apps will need more than two test cases, so parametrizing seems like a good path forward in any case. I'm now questioning whether the entire response structure needs to be tested against, but I decided to keep that aspect of the tests in the interest of keeping the commit slim. Also opted to leave out fixtures to limit scope. Curious to hear your thoughts.